Library/Homebrew/dependencies.rb

```diff
@@ -123,3 +123,61 @@ def reject_ignores(dependables, ignores, includes)
     end
   end
 end
+
+# An adapter for casks to provide dependency information in a formula-like interface
+class CaskDependent
+  def initialize(cask)
+    @cask = cask
+  end
+
+  def name
+    @cask.token
+  end
+
+  def full_name
+    @cask.full_name
+  end
+
+  def runtime_dependencies
+    recursive_dependencies
+  end
+
+  def deps
+    @deps ||= begin
+      @cask.depends_on.formula.map do |f|
+        Dependency.new f
+      end
+    end
+  end
+
+  def requirements
+    @requirements ||= begin
+      requirements = []
+      dsl_reqs = @cask.depends_on
+
+      dsl_reqs.arch&.each do |arch|
+        requirements << ArchRequirement.new([:x86_64]) if arch[:bits] == 64
+        requirements << ArchRequirement.new([arch[:type]])
```

Comment:
Not sure if this is correct on my half? Based on `ArchRequirement::satisfy`, seems like `:x86_64` means 64-bit, but with arm macs right around the corner is this still correct?

Reply:
I don't think we have a single cask using `depends_on arch:` so not sure what this should look like.

Reply:
```
Casks/locklizard-safeguard-viewer.rb
12:  depends_on arch: ":x86_64"
```

That's the only one. No `bits` references. I think you can punt on adding this architecture requirement for now, maybe?

Reply:
> That's the only one.

And that one is even using it wrong with a string instead of symbol.